doc-content/enterprise-only/openshift/githooks-proc.adoc

```diff
@@ -0,0 +1,80 @@
+----
++
+.. Mount the secret in the ssh key path of the {CENTRAL} deployment:
++
+[subs="attributes,verbatim,macros"]
+----
+oc set volume dc/<myapp>-{PRODUCT_INIT}centr --add --type secret --secret-name {PRODUCT_INIT}-centr-gitkey --mount-path=/home/jboss/.ssh --name=ssh-key
+----
++
+Where `<myapp>` is the application name that was set when configuring the template. 
++
+. Create the Git hooks directory. For instructions, see the https://git-scm.com/docs/githooks[Git hooks reference documentation].
++
+For example, a simple git hooks directory can provide a post-commit hook that pushes the changes upstream. If the project was imported into {CENTRAL} from a repository, this repository remains configured as the upstream repository. Create a file named `post-commit` with permission values `755` and the following content:
++
+[subs="attributes,verbatim,macros"]
+----
+git push
+----
++
+. Supply the Git hooks directory to the {CENTRAL} deployment. You can use a configuration map or a persistent volume.
+.. If the Git hooks consist of one or several fixed script files, use a configuration map. Complete the following steps:
+... Change into the Git hooks directory that you have created.
+... Create an OpenShift configuration map from the files in the directory. Run the following command:
++
+[subs="attributes,verbatim,macros"]
+----
+oc create configmap git_hooks --from-file=<file_1>=<file_1> --from-file=<file_2>=<file_2> ...
+----
++
+Where `file_1`, `file_2` and so on are git hook script files. For example:
++
+[subs="attributes,verbatim,macros"]
+----
+oc create configmap git_hooks --from-file=post-commit=post-commit
+----
++
+... Mount the configuration map on the {CENTRAL} deploymentm in the path that you have configured:
++
+[subs="attributes,verbatim,macros"]
+----
+oc set volume dc/<myapp>-{PRODUCT_INIT}centr --add --type configmap --configmap-name git_hooks  --mount-path=<git_hooks_dir> --name=git_hooks
+----
++
+Where `<myapp>` is the application name that was set when configuring the template and `<git_hooks_dir>` is the value of `GIT_HOOKS_DIR` that was set when configuring the template.
++
+.. If the Git hooks consist of many files or require writing any files at runtime, use a persistence volume. You must create a persistent volume, create a persistent volume claim and associate the volume with the claim, transfer files to the volume, and mount the volume in the `_myapp_-{PRODUCT_INIT}centr` deployment config (where _myapp_ is the application name). For instructions about creating and mounting persistence volumes, see https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/developer_guide/dev-guide-persistent-volumes[Using persistent volumes]. For instructions about copying files onto a persistent volume, see https://blog.openshift.com/transferring-files-in-and-out-of-containers-in-openshift-part-3/[Transferring files in and out of containers].
```

Comment:
Git hooks shouldn't need to be modified at runtime. I would mention the case where githooks are long scripts or depend on binaries (executables, jars, etc.). There, it is also advisable to use persistent volume claims.

Reply:
Maybe the files are out of sync. I find this in the doc:
If the Git hooks consist of long files or depends on binaries, sucn as executable or KJAR files, use a persistence volume. You must create a persistent volume, create a persistent volume claim and associate the volume with the claim, transfer files to the volume, and mount the volume in the myapp-rhpamcentr deployment config (where myapp is the application name). For instructions about creating and mounting persistence volumes, see Using persistent volumes. For instructions about copying files onto a persistent volume, see Transferring files in and out of containers.

or depend (singular)
sucn (typo)
config - configuration